docs/TablePartitionsStorageGroups.md

```diff
@@ -0,0 +1,421 @@
+- All archives serving the storage group are damaged or lost.
+
+A storage group that has gone permanently offline must be deleted.
+Deleting a storage group informs the database that it no longer needs to wait for the storage group to come back online.
+
+>**Note**
+> A storage group can only be deleted from the database if it was previously removed from all archives or if the archives serving the storage group are deleted.
+
+Remove the Helm release corresponding to this storage group.
+
+```sh
+helm delete sgeast
+```
+
+Delete all _archive_ Persistent Volume Claims (PVCs) for this storage group.
+
+```sh
+for pvc in $(kubectl get pvc -l domain=nuodb,database=demo,release=sgeast -oname | grep '/archive-volume-'); do
+    kubectl delete $pvc
+done
+```
+
+If [Kubernetes Aware Admin][1] (KAA) is configured, it will automatically delete all archive objects from the domain state that corresponds to the deleted PVCs.
+
+Delete a permanently offline storage group from the database.
+
+>**Warning**
+> Deleting a storage group will remove all table partitions and data stored in the storage group.
+
+```sh
+kubectl exec admin-nuodb-cluster0-0 -- \
+    nuocmd delete storage-group --db-name demo --sg-name sgeast
+```
+
+### Storage Group Maintenance
+
+A storage group goes temporarily offline for the following reasons:
+
+- There is no `RUNNING` SM servicing this storage group.
+- None of the SMs servicing the storage group can become `RUNNING` and maintenance operations must be performed such as `nuoarchive check`.
+
+A storage group that is temporarily offline must be brought into maintenance.
+This will inform TEs and scheduled backups that they no longer need to wait for the storage group to come online.
+
+>**Important**
+> Bringing a storage group into maintenance will exclude it from database backups. Restoring from such a backup could cause data loss or inconsistent data and must be used with caution.
+
+To put a storage group into maintenance, scale down all corresponding SMs and remove all the archives that service it.
+
+```sh
+kubectl scale statefulset -l domain=nuodb,database=demo,release=sgeast --replicas 0
+```
+
+Get the archive IDs that are associated with the storage group.
+The `archive_states` property has all archive IDs associated to this storage group.
+
+```sh
+kubectl exec admin-nuodb-cluster0-0 -- \
+    nuocmd get storage-groups --db-name demo
+```
+
+If [Kubernetes Aware Admin][1] is configured, there will be only one archive associated with the storage group as KAA will automatically remove archives for scaled-down SMs.
+
+```text
+...
+StorageGroup(archive_states={0: ADDED}, db_name=demo, id=10, leader_candidates=[], name=SGEAST, process_states={}, state=Unavailable)
+...
+```
+
+Delete, _but not_ purge, the last archive that services this storage group.
+
+```sh
+kubectl exec admin-nuodb-cluster0-0 -- \
+    nuocmd delete archive --archive-id 0
```

Comment:
Is this actually necessary? I know that a database cannot be left without archive objects, but there is nothing in the nuoadmin code that prevents the final archive for a storage group other than UNPARTITIONED from being removed. Is this enforced somewhere else, e.g. KAA?

Reply:
The last archive for a storage group won't be automatically deleted (there is an enforcement in KAA). It should be deleted manually so that `nuodocker` doesn't wait for that storage group to become _available_ anymore on the database cold start. A storage group without archives will be excluded from the covering set of storage groups for the database during hot-copy as well.